### PR TITLE
integrating with widget loader

### PR DESCRIPTION
**Pull request by leonoragilmour**
Adding the fields as dependents on available post types and widgets (using widget loader)

Corresponding Croissant Heavy PR
https://github.com/shortlist-digital/croissant-heavy/pull/48

Corresponding branch on Agreable Base Theme
https://github.com/shortlist-digital/agreable-base-theme/tree/feature/post-type-picker

### CODE REVIEW

**Comment by leonoragilmour** [comment]
[Creating a post type picker plugin so widgets can be applied to different post types, brand dependent](https://trello.com/c/zOqxX9To/209-creating-a-post-type-picker-plugin-so-widgets-can-be-applied-to-different-post-types-brand-dependent)

File: app/CustomFields/PostTypePickerFields.php [hunk 1]
```diff
@@ -0,0 +1,60 @@
+<?php
+
+namespace WpWidgetPostTypePickerPlugin\CustomFields;
+
+include __DIR__."/../../../../themes/agreable-base-theme/custom-fields/WidgetLoader.php";
```
Comment (ecoad) [inline app/CustomFields/PostTypePickerFields.php]: You can use https://codex.wordpress.org/Function_Reference/get_stylesheet_directory to get the base theme location

Comment (leonoragilmour) [inline app/CustomFields/PostTypePickerFields.php]: wow that would've saved a lot of headache 🙄 ... will update tomorrow

File: app/CustomFields/PostTypePickerFields.php [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+class PostTypePickerFields {
+
+	public function init() {
```
Comment (ecoad) [inline app/CustomFields/PostTypePickerFields.php]: I don't think this function should include `add_action('init', ...)`. The behaviour is more predictable if whenever you call init, it always does init stuff. So lets move the `add_action` out of here and place it before it is called down in `add_custom_fields() { ... }`. It also then makes this function more testable if we were to write any tests for it.

File: app/CustomFields/PostTypePickerFields.php [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+				$widgets[$widget['key']] = $widget['name'];
+			};
+
+			$post_types = get_post_types(array('public' => true)); // get post types
+			$post_types["category"] = "category";
+			unset($post_types["attachment"]);
+
+			foreach ($widgets as $key=>$widget) {
+				$field = array (
+					'key' => $key . '_available_post_types',
```
Comment (ecoad) [inline app/CustomFields/PostTypePickerFields.php]: The naming of the key for fields should match the group key.

Currently your group key is:
`widget_post_types_picker `

Then the fields are:
`<widget name>_available_post_types`

The fields should actually be:
`widget_post_types_picker _<widget name>_available`

File: app/CustomOptions/WidgetOptions.php [hunk 4]
```diff
@@ -0,0 +1,21 @@
+<?php
+
+namespace WpWidgetPostTypePickerPlugin\CustomOptions;
+
+class WidgetOptions {
+
+	public function init() {
+		add_action('init', function() {
```
Comment (ecoad) [inline app/CustomOptions/WidgetOptions.php]: Move `add_action` out of here too please

File: app/CustomFields/PostTypePickerFields.php (hunk 2 above)
Comment (leonoragilmour) [inline app/CustomFields/PostTypePickerFields.php]: might need some help with this tomorrow as I can't get it to work if I move it around...

**Comment by leonoragilmour** [comment]
@ecoad any new thoughts?

**Pull request merged**